common/src/test/java/tachyon/security/LoginModuleTest.java

```diff
@@ -55,4 +64,47 @@ public void simpleLoginTest() throws Exception {
   }
 
   // TODO: Kerberos login test
+
+  /**
+   * Test whether we can get login user with conf in SIMPLE mode.
+   * @throws Exception
+   */
+  @Test
+  public void getSimpleLoginUserTest() throws Exception {
+    TachyonConf conf = new TachyonConf();
+
+    // set conf to "SIMPLE" and get login user
+    conf.set(Constants.TACHYON_SECURITY_AUTHENTICATION, "SIMPLE");
+    UserInformation.setTachyonConf(conf);
+
+    UserInformation loginUser = UserInformation.getTachyonLoginUser();
+
+    Assert.assertFalse(loginUser.getUserName().isEmpty());
+  }
+
+  /**
+   * Test initializing with conf in NOSASL mode.
+   * @throws Exception
+   */
+  @Test
+  public void initConfTest() throws Exception {
+    // throw exception when initializing conf with "NOSASL"
+    TachyonConf conf = new TachyonConf();
+    if (conf.get(Constants.TACHYON_SECURITY_AUTHENTICATION, "").equalsIgnoreCase("NOSASL")) {
+      try {
+        UserInformation.getTachyonLoginUser();
+        Assert.fail();
+      } catch (UnsupportedOperationException e) {
+        Assert.assertTrue(e.getMessage().contains("not supported in NOSASL mode"));
+      }
+    } else {
+      conf.set(Constants.TACHYON_SECURITY_AUTHENTICATION, "NOSASL");
+      try {
+        UserInformation.setTachyonConf(conf);
+        Assert.fail();
+      } catch (UnsupportedOperationException e) {
```

Comment:
We are now converting to use the way JUNIT 4 is providing to test exception. Please find examples here:

```
Thanks for adding unit test covering cases with exceptions thrown! this is really helpful for ensuring Tachyon gets solid components.

A style issue here. Tachyon is using JUNIT4 and given this version is used, we use the following convention to unit test thrown exceptions:
http://junit.org/apidocs/org/junit/rules/ExpectedException.html
And here is an example we unit tests with exceptions in Tachyon
https://github.com/amplab/tachyon/blob/master/servers/src/test/java/tachyon/worker/block/BlockMetadataManagerViewTest.java#L110.

```